wolfcrypt/src/asn.c

```diff
@@ -2886,7 +2919,19 @@ int wc_CreatePKCS8Key(byte* out, word32* outSz, byte* key, word32 keySz,
         if (key == NULL || out == NULL || outSz == NULL) {
             return BAD_FUNC_ARG;
         }
-
+        
+        /* check if the key has pkcs8 header                                   */
```

Comment:
If the key already has the PKCS#8 header, wouldn't it make sense to return success early, since there's no work to be done?

Reply:
It needs to create key body for the caller.